### ISSUE TITLE
[Bug]: source code content was incorrectly replaced by file hash name.

### ISSUE THREAD

**Issue by bigbossx** [web-infra-dev/rspack#8474]
### System Info

rsbuild version: 1.0.19

```js
// use in project
export const lazyLoadDateHolidays = () => {
  return import(
    /* webpackChunkName: "date-holidays"*/
    /* webpackPrefetch: true */
    'date-holidays'
  );
};

```

break down in production build

![Image](https://github.com/user-attachments/assets/122c689d-ff00-40c5-b1ca-d2aad41e916a)

### Details

- is it stable reproduce ? No,  i can't provider a mini reproduce

- webpack has some problem？No, only happen with rsbuild 

maybe rspack need a better way to replace hash content in chunk, not global replace

please cc @chenjiahan btw, thanks for your quick answer in wechat. help me a lot

### Reproduce Steps

[source code was break down in](https://github.com/commenthol/astronomia/blob/6374bf5841c474bb3d5f6f83aded4940331d0207/data/vsop87Bearth.js#L622)

![Image](https://github.com/user-attachments/assets/0479b76c-4349-48e9-844b-e9f5559af636)

**Comment by bigbossx** [comment]
workaround:

```ts
// rsbuild.config.ts

export default defineConfig({
   output:{
     filenameHash: 'contenthash:16' // long hash name to reduce the probability of conflict
   }
})
```

**Comment by chenjiahan** [comment]
Yes, we have discussed this issue. Using a content hash of 10 characters or more should avoid this problem, but it will also increase the bundle size slightly.

If Rspack can implement a more secure hash replacement, that will be the best approach.

**Comment by chenjiahan** [comment]
Duplicate with: https://github.com/web-infra-dev/rspack/issues/5339

**Comment by bigbossx** [comment]
just a suggestion:

1. maybe we can change the [default config in rsbuild](https://github.com/web-infra-dev/rsbuild/blob/a5a73616006f718a22dd16cf04bf566827c5a356/packages/core/src/helpers/index.ts#L233) to
```js
const getHash = () => {
    if (typeof filenameHash === 'string') {
      return filenameHash ? `.[${filenameHash}]` : '';
    }
    return filenameHash ? '.[contenthash:12]' : '';
};
```

2. and note this `warning or tips` in [rspack docs](https://rspack.dev/config/output#outputchunkfilename) 

in next release

because it seems like it will take some time to really fix it, but before that, we can reduce the probability of bugs happening

**Comment by sibelius** [comment]
does

 /* webpackChunkName: "[request]" */

works ?

**Comment by Benno-Wu** [comment]
i find a work around: [RealContentHashCheckPlugin](https://github.com/Benno-Wu/RealContentHashCheckPlugin)
which helps on detecting the hash collision
but still need to change the source code
